bench/workgen/workgen.cxx

```diff
@@ -480,42 +469,89 @@ int Monitor::run() {
-                << "," << interval.read.min_latency
-                << "," << interval.read.max_latency
-                << "," << interval.insert.average_latency()
-                << "," << interval.insert.min_latency
-                << "," << interval.insert.max_latency
-                << "," << interval.update.average_latency()
-                << "," << interval.update.min_latency
-                << "," << interval.update.max_latency
-                << std::endl;
-
-        if (_json != NULL) {
-#define    WORKGEN_TIMESTAMP_JSON        "%Y-%m-%dT%H:%M:%S"
-            buf_size = strftime(time_buf, sizeof(time_buf),
-              WORKGEN_TIMESTAMP_JSON, tm);
-            ASSERT(buf_size <= sizeof(time_buf));
-            snprintf(&time_buf[buf_size], sizeof(time_buf) - buf_size,
-              ".%3.3" PRIu64 "Z", (uint64_t)ns_to_ms(t.tv_nsec));
-
-            // Note: we could allow this to be configurable.
-            int percentiles[4] = {50, 95, 99, 0};
+void Monitor::_format_out_header() {
+    (*_out) << "#time,"
+            << "totalsec,"
+            << "read ops per second,"
+            << "insert ops per second,"
+            << "update ops per second,"
+            << "checkpoints,"
+            << "read average latency(uS),"
+            << "read minimum latency(uS),"
+            << "read maximum latency(uS),"
+            << "insert average latency(uS),"
+            << "insert min latency(uS),"
+            << "insert maximum latency(uS),"
+            << "update average latency(uS),"
+            << "update min latency(uS),"
+            << "update maximum latency(uS)"
+            << std::endl;
+}
+
+void Monitor::_format_out_entry(Stats *pinterval, double interval_secs, timespec *ptimespec, bool checkpointing, tm *tm) {
+    char time_buf[64];
+    uint64_t cur_reads = (uint64_t)(pinterval->read.ops / interval_secs);
+    uint64_t cur_inserts = (uint64_t)(pinterval->insert.ops / interval_secs);
+    uint64_t cur_updates = (uint64_t)(pinterval->update.ops / interval_secs);
+    uint64_t totalsec = ts_sec((*ptimespec) - _wrunner._start);
+
+    (void)strftime(time_buf, sizeof(time_buf), "%b %d %H:%M:%S", tm);
+    (*_out) << time_buf
+            << "," << totalsec
+            << "," << cur_reads
+            << "," << cur_inserts
+            << "," << cur_updates
+            << "," << (checkpointing ? 'Y' : 'N')
+            << "," << pinterval->read.average_latency()
+            << "," << pinterval->read.min_latency
+            << "," << pinterval->read.max_latency
+            << "," << pinterval->insert.average_latency()
+            << "," << pinterval->insert.min_latency
+            << "," << pinterval->insert.max_latency
+            << "," << pinterval->update.average_latency()
+            << "," << pinterval->update.min_latency
+            << "," << pinterval->update.max_latency
+            << std::endl;
+}
 
+void Monitor::_format_json_prefix(char *version) {
```

Comment:
We could use const std::string &version here

Reply:
Would `const char *version` be enough? std::string would allocate extra memory, not a big deal but still.

Reply:
Changed in the latest commit. Thanks.